packages/cosmos_ui_components/lib/components/gradient_avatar.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:string_to_hex/string_to_hex.dart';
+
+class GradientAvatar extends StatelessWidget {
+  final String stringKey;
+
+  const GradientAvatar({Key? key, required this.stringKey}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = generateColorsFromAddress(stringKey);
+    return CircleAvatar(
+      child: Container(
+        decoration: BoxDecoration(
+          shape: BoxShape.circle,
+          gradient: LinearGradient(colors: [colors[0], colors[1]]),
+        ),
+      ),
+    );
+  }
+
+  List<Color> generateColorsFromAddress(String address) {
+    final color1 = StringBuffer();
+    final color2 = StringBuffer();
+
+    color1.write('#');
+    color2.write('#');
+
+    // Initialised it here because linter was warning me not to declare multiple variables in single line
+    var j = address.length - 7;
```

Comment:
let's use more descriptive names for the variables. also the comment above is not needed, variables should always be initialized in separate lines for readability